spring-integration-core/src/main/java/org/springframework/integration/dsl/StandardIntegrationFlow.java

```diff
@@ -74,8 +79,31 @@
 
 	private boolean running;
 
+	@SuppressWarnings("unchecked")
 	StandardIntegrationFlow(Map<Object, String> integrationComponents) {
 		this.integrationComponents = new LinkedHashMap<>(integrationComponents);
+		Iterator<Object> iterator = this.integrationComponents.keySet().iterator();
+		Object one = null;
+		Object two = null;
+		Object three = null;
+		if (iterator.hasNext()) {
+			one = iterator.next();
+		}
+		if (iterator.hasNext()) {
+			two = iterator.next();
+		}
+		if (iterator.hasNext()) {
+			three = iterator.next();
+		}
+		// We have at least 4 components and the first three are a synthesized bridge
+		if (iterator.hasNext() && three != null
+				&& one instanceof DirectChannel && ((DirectChannel) one).getBeanName() == null
+				&& two instanceof GenericEndpointSpec
+				&& ((GenericEndpointSpec<? extends MessageHandler>) two).getHandler() instanceof BridgeHandler
```

Comment:
I may intentionally want a bridge between channels. Even if they are `MessageChannelReference` instances, I may add `ChannelInterceptor` to them eventually at runtime.